src/lambdas/deleteUser/deleteUser.js

```diff
@@ -0,0 +1,50 @@
+const DynamoDB = require('aws-sdk/clients/dynamodb');
+const db = new DynamoDB.DocumentClient();
+const clone = require('clone');
+const Log = require('@dazn/lambda-powertools-logger');
+const TABLE_NAME = process.env.TABLE_NAME || '';
+
+module.exports.handler = async event => {
+    try {
+        const { id: userId } = event.pathParameters || {};
+        if (!userId) throw 'no user id';
+
+        // get item to delete
+        const originalParams = { TableName: TABLE_NAME, Key: { pk: userId, sk: `USER` } };
+        const doc = await db.get(originalParams).promise();
+        const { Item } = doc;
+        if (!Item) throw 'user does not exist';
+
+        // clone a backup item
+        const backupItem = clone(Item);
+        backupItem.sk = 'DELETED_USER';
+        backupItem.updatedAt = new Date().toISOString();
+
+        // store the backup
+        const backupParams = { TableName: TABLE_NAME, Item: backupItem };
+        await db.put(backupParams).promise();
+
+        //  delete the original
+        await db.delete(originalParams).promise();
```

Comment:
How would you go about building an audit log of all delete requests and capturing who initiated it as well as other details?

Reply:
I like the idea of monitoring the Dynamo stream for changes like this so we don't tie up the delete lambda with this extra work.  We could also move the entire delete request to an SQS queue, immediately returning a 202 status code (Accepted), and then have the lambda processing the queue and record the audit data.